synedrion/src/tools/secret.rs

```diff
@@ -79,315 +73,222 @@ impl<T: Zeroize> Debug for Secret<T> {
-        iter.fold(Secret::<T>::default(), |accum, x| accum + x)
-    }
-}
-
-impl Secret<Scalar> {
-    pub fn mul_by_generator(&self) -> Point {
-        self.expose_secret().mul_by_generator()
-    }
-
-    pub fn invert(&self) -> Option<Secret<Scalar>> {
-        Secret::maybe_init_with(|| Option::from(self.expose_secret().invert()))
-    }
-}
-
-impl Mul<Secret<Scalar>> for Point {
-    type Output = Point;
-    fn mul(self, scalar: Secret<Scalar>) -> Self::Output {
-        self * scalar.expose_secret()
-    }
-}
-
-impl Mul<&Secret<Scalar>> for Point {
-    type Output = Point;
-    fn mul(self, scalar: &Secret<Scalar>) -> Self::Output {
-        self * scalar.expose_secret()
-    }
-}
-
-impl Mul<Secret<Scalar>> for &Point {
-    type Output = Point;
-    fn mul(self, scalar: Secret<Scalar>) -> Self::Output {
-        self * scalar.expose_secret()
-    }
-}
-
 impl<T: Zeroize + Retrieve<Output: Zeroize + Clone>> Retrieve for Secret<T> {
     type Output = Secret<T::Output>;
     fn retrieve(&self) -> Self::Output {
         Secret::init_with(|| self.expose_secret().retrieve())
     }
 }
 
-impl<T: Zeroize> Secret<T> {
-    pub fn pow_bounded<V>(&self, exponent: &Bounded<V>) -> Self
-    where
-        T: Exponentiable<V>,
-        V: Integer + crypto_bigint::Bounded + Encoding + ConditionallySelectable,
-    {
-        // TODO: do we need to implement our own windowed exponentiation to hide the secret?
-        Secret::init_with(|| self.expose_secret().pow_bounded(exponent))
-    }
-
-    pub fn pow_signed_vartime<V>(&self, exponent: &Signed<V>) -> Self
+impl<T: Zeroize + Clone> Secret<T> {
+    pub fn pow<V>(&self, exponent: &V) -> Self
     where
         T: Exponentiable<V>,
-        V: Integer + crypto_bigint::Bounded + Encoding + ConditionallySelectable,
     {
         // TODO: do we need to implement our own windowed exponentiation to hide the secret?
         // The exponent will be put in a stack array when it's decomposed with a small radix
         // for windowed exponentiation. So if it's secret, it's going to leave traces on the stack.
         // With the multiplication, for example, there's less danger since Uints implement *Assign traits which we use,
         // so theoretically anything secret will be overwritten.
-        Secret::init_with(|| self.expose_secret().pow_signed_vartime(exponent))
+        Secret::init_with(|| self.expose_secret().pow(exponent))
     }
 }
 
 impl<T: Zeroize + Integer<Monty: Zeroize>> Secret<T> {
     pub fn to_montgomery(&self, params: &<T::Monty as Monty>::Params) -> Secret<T::Monty> {
         // `self` has to be cloned and passed by value, which means it may be retained on the stack.
         // Can't help it with the current `Monty::new()` signature.
+
+        // NOTE: `params` is cloned here and can remain on the stack.
+        // See https://github.com/RustCrypto/crypto-bigint/issues/704;
+        // after that's fixed, we can properly treat secret `params`.
```

Comment:
https://github.com/RustCrypto/crypto-bigint/issues/704 is fixed now; do we need to update this comment perhaps?

Reply:
Removed the links, kept just the TODO for #162